internals/overlord/state/notices_test.go

```diff
@@ -408,6 +408,28 @@ func (s *noticesSuite) TestWaitNoticesTimeout(c *C) {
 	c.Assert(notices, HasLen, 0)
 }
 
+func (s *noticesSuite) TestReadStateWaitNotices(c *C) {
+	st := state.New(nil)
+	st.Lock()
+	defer st.Unlock()
+
+	marshalled, err := st.MarshalJSON()
+	c.Assert(err, IsNil)
+
+	buf := bytes.NewBuffer(marshalled)
+
+	st2, err := state.ReadState(nil, buf)
```

Comment:
Nit: having this a variable doesn't add value. Suggestion:

```suggestion
	st2, err := state.ReadState(nil, bytes.NewBuffer(marshalled))
```